lib/Dialect/ICE40/ICE40Ops.cpp

```diff
@@ -0,0 +1,44 @@
+//===- ICE40Ops.cpp - Implement the ICE40 operations ----------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements ICE40 ops.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/ICE40/ICE40Ops.h"
+#include "mlir/IR/DialectImplementation.h"
+#include "mlir/IR/ImplicitLocOpBuilder.h"
+
+using namespace circt;
+using namespace ice40;
+
+//===----------------------------------------------------------------------===//
+// SBLut4Op
+//===----------------------------------------------------------------------===//
+
+LogicalResult SBLut4Op::verify() { return success(); }
```

Comment:
This is really exciting! One nit - is it worth verifying that `lut_init` has 16 elements? Or is this just a placeholder verifier

Reply:
Agreed it should check.